### PR TITLE
test: add e2e to clear privacy data

### PR DESCRIPTION
**Pull request by cortisiko**
## **Description**

In an effort to increase our overall e2e test coverage, this PR aims to add E2E coverage for the clear privacy flow. The test steps are as follows:
- Launch app connected to test dapp
- Verify that the account is connected to test dapp
- Go to privacy and security and clear privacy data
- Verify that account is no longer connected to test dapp

## **Related issues**

Fixes:

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: app/components/Views/Settings/SecuritySettings/Sections/ClearPrivacy/ClearPrivacy.tsx [hunk 1]
```diff
@@ -40,7 +43,7 @@ const ClearPrivacy = () => {
       onRequestClose={() => setModalVisible(false)}
       onConfirmPress={clearApprovals}
     >
-      <View style={styles.modalView}>
+      <View style={styles.modalView} testID={ClearPrivacyModalSelectorsIDs.CONTAINER}>
```
Comment (cortisiko) [inline app/components/Views/Settings/SecuritySettings/Sections/ClearPrivacy/ClearPrivacy.tsx:46]: Adding test ID to clear privacy modal

File: app/components/Views/Settings/SecuritySettings/Sections/ClearPrivacy/ClearPrivacy.tsx [hunk 2]
```diff
@@ -66,6 +69,7 @@ const ClearPrivacy = () => {
       <View style={styles.accessory}>
         <Button
           variant={ButtonVariants.Secondary}
+          testID={SecurityPrivacyViewSelectorsIDs.CLEAR_PRIVACY_DATA_BUTTON}
```
Comment (cortisiko) [inline app/components/Views/Settings/SecuritySettings/Sections/ClearPrivacy/ClearPrivacy.tsx:72]: Adding testID to clear Privacy button

File: e2e/pages/Settings/SecurityAndPrivacy/ClearPrivacyModal.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
+import {
```
Comment (cortisiko) [inline e2e/pages/Settings/SecurityAndPrivacy/ClearPrivacyModal.js:1]: page object for clear privacy modal

File: e2e/pages/Settings/SecurityAndPrivacy/SecurityAndPrivacyView.js [hunk 4]
```diff
@@ -18,6 +18,11 @@ class SecurityAndPrivacy {
       SecurityPrivacyViewSelectorsIDs.REVEAL_SEED_BUTTON,
     );
   }
+  get clearPrivacyDataButton() {
```
Comment (cortisiko) [inline e2e/pages/Settings/SecurityAndPrivacy/SecurityAndPrivacyView.js:21]: element locator getting to find clear privacy data button in the privacy and security settings

File: e2e/pages/Settings/SecurityAndPrivacy/SecurityAndPrivacyView.js [hunk 5]
```diff
@@ -144,6 +149,13 @@ class SecurityAndPrivacy {
     );
   }
 
+  async scrollToClearPrivacyData() {
```
Comment (cortisiko) [inline e2e/pages/Settings/SecurityAndPrivacy/SecurityAndPrivacyView.js:152]: Scroll method to get to the clear privacy data button

File: e2e/pages/Settings/SecurityAndPrivacy/SecurityAndPrivacyView.js [hunk 6]
```diff
@@ -165,6 +177,9 @@ class SecurityAndPrivacy {
   async tapMetaMetricsToggle() {
     await Gestures.waitAndTap(this.metaMetricsToggle);
   }
+  async tapClearPrivacyData() {
```
Comment (cortisiko) [inline e2e/pages/Settings/SecurityAndPrivacy/SecurityAndPrivacyView.js:180]: method to tap on clear privacy data

File: e2e/selectors/Settings/SecurityAndPrivacy/ClearPrivacyModal.selectors.js [hunk 7]
```diff
@@ -0,0 +1,10 @@
+import enContent from '../../../../locales/languages/en.json';
```
Comment (cortisiko) [inline e2e/selectors/Settings/SecurityAndPrivacy/ClearPrivacyModal.selectors.js:1]: selector object for the clear privacy modal

File: e2e/selectors/Settings/SecurityAndPrivacy/SecurityPrivacyView.selectors.js [hunk 8]
```diff
@@ -11,7 +11,8 @@ export const SecurityPrivacyViewSelectorsIDs = {
   REMEMBER_ME_TOGGLE: 'turn-on-remember-me',
   SHOW_PRIVATE_KEY: 'show-private-key',
   BIOMETRICS_TOGGLE: 'biometrics-option',
-  DEVICE_PASSCODE_TOGGLE: 'DevicePasscodeOption',
+  DEVICE_PASSCODE_TOGGLE: 'device-passcode-option',
```
Comment (cortisiko) [inline e2e/selectors/Settings/SecurityAndPrivacy/SecurityPrivacyView.selectors.js:14]: clean up this selector string to conform to best practice doc

File: e2e/selectors/Settings/SecurityAndPrivacy/SecurityPrivacyView.selectors.js [hunk 9]
```diff
@@ -11,7 +11,8 @@ export const SecurityPrivacyViewSelectorsIDs = {
...
+  CLEAR_PRIVACY_DATA_BUTTON: 'clear-privacy-data-button',
```
Comment (cortisiko) [inline e2e/selectors/Settings/SecurityAndPrivacy/SecurityPrivacyView.selectors.js:15]: new selector string for the clear privacy button

File: e2e/tags.js [hunk 10]
```diff
@@ -1,13 +1,13 @@
...
-  regression: 'Regression',
-  smokeAccounts: 'SmokeAccounts',
-  smokeCore: 'SmokeCore',
-  smokeConfirmations: 'SmokeConfirmations',
-  SmokeSwaps: 'SmokeSwaps',
-  SmokeRest: 'SmokeRest',
-  smokeAssets: 'smokeAssets',
-  smokeIdentity: 'smokeIdentity',
-  smokeMultiChain: 'SmokeMultiChain',
+  regression: 'Regression:',
```
Comment (cortisiko) [inline e2e/tags.js:2]: adding ":" to improve test case readability.

**Review by EtherWizard33 — APPROVED** [review]
LGTM!

**Review by Andepande — APPROVED** [review]
LGTM

**Pull request merged**